packages/dds/map/src/directory.ts

```diff
@@ -1304,9 +1315,10 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 		private readonly runtime: IFluidDataStoreRuntime,
 		private readonly serializer: IFluidSerializer,
 		public readonly absolutePath: string,
-		private readonly logger: ITelemetryLoggerExt,
+		private readonly logger: ITelemetryBaseLogger,
 	) {
 		super();
+		this.loggerExt = createChildLogger({ logger: this.logger });
```

Comment:
it would be nice if there was a way to upcast, as most the time this should be a ext, something like toTelemetryLoggerEx(logger): ITelemetryLoggerBase, that inspects the object, and determines if it already is one, and if not createChildLogger